### PR TITLE
chore(ci): add golangci linter

### PR DESCRIPTION
**Pull request by kaminek**
Signed-off-by: Amine Kherbouche <kaminek92@gmail.com>

### CODE REVIEW

**Comment by moitias** [comment]
:raised_hands: LGTM :+1:

**Review by moitias — APPROVED** [review]

**Review by moitias — CHANGES_REQUESTED** [review]
We should at least;

- enable go fmt in golangci-lint
- add pre-commit hooks that run the same linters

before merging

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.37.1
```
Comment (scop) [inline .github/workflows/lint.yml:14]: If there's no reason to install an older one:

```suggestion
          version: v1.40.0
```

**Comment by scop** [comment]
>     * enable go fmt in golangci-lint

Tested adding `gofmt` to the config locally, it resulted in no new findings. But agree we should add it if there's no strong/particular reason not to.

**Review by moitias — APPROVED** [review]

**Pull request merged**